app/src/test/java/ch/epfl/sweng/swenggolf/FireDatabaseTest.java

```diff
@@ -241,6 +244,38 @@ public Void answer(InvocationOnMock invocation) {
         };
     }
 
+    @NonNull
+    private Answer<Void> setUpFollowersDataSnapshot(Map<String, List<String>> usersFollowing) {
+        List<DataSnapshot> children = new ArrayList<>();
+
+        for (String userId : usersFollowing.keySet()) {
+            List<DataSnapshot> grandChildren = new ArrayList<>();
+            for (String follower : usersFollowing.get(userId)) {
+                DataSnapshot grandChild = mock(DataSnapshot.class);
+                when(grandChild.getKey()).thenReturn(follower);
+                grandChildren.add(grandChild);
+            }
+
+            DataSnapshot child = mock(DataSnapshot.class);
+            when(child.getKey()).thenReturn(userId);
+            when(child.getChildren()).thenReturn(grandChildren);
+            children.add(child);
+        }
+
+        final DataSnapshot parent = mock(DataSnapshot.class);
+        when(parent.getChildren()).thenReturn(children);
+        when(parent.exists()).thenReturn(true);
+
+        return new Answer<Void>() {
+            @Override
+            public Void answer(InvocationOnMock invocation) {
+                ValueEventListener listener = invocation.getArgument(0);
+                listener.onDataChange(parent);
+                return null;
```

Comment:
I think this return null is unnecessary but since it just explicits the implicit behavior of the method guess we'll leave it this way.

Reply:
I just did the same way as what has been done before